include/private/UploadArtifactTask.h

```diff
@@ -14,15 +14,15 @@ namespace prism
 namespace connect
 {
 
-class ArtifactUploadHelper;
+class Client;
 
 class UploadArtifactTask
 {
 public:
     virtual ~UploadArtifactTask()
     {}
 
-    virtual Status execute(ArtifactUploadHelper* uploader) const = 0;
+    virtual Status execute(Client& client, id_t accountId, id_t cameraId) const = 0;
```

Comment:
Client has methods that doesn't need one or both of IDs. 
Uploader indeed internally has both IDs and now it passes them into execute() method.
Options I see here:
1. Pass accountId and cameraId into UploadAritfactTask* constructor. See no cons with it.
2. Aggregate client and both IDs into a struct with no methods (to avoid creating method in struct for each method in client). Kind of one more level of indirection that will result in a single parameter to execute(). Not sure it makes much sense.
3. Client wrapper with both IDs provided on construction. Cons I see is we have one more class and we need to implement wrapper method for each corresponding client method, this is what I'm trying to avoid.

So I will do it per 1.